xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-ui/src/main/java/org/xwiki/test/ui/po/BasePage.java

```diff
@@ -693,6 +705,60 @@ private void useShortcutKeyAndLoads(CharSequence... shortcut)
         getDriver().waitUntilPageIsReloaded();
     }
 
+    /**
+     * Validate WCAG properties on the current base page.
+     * Default behavior is to check the cache.
+     * @param wcagContext the current wcag context
+     */
+    protected void validateWCAG(WcagContext wcagContext)
+    {
+        validateWCAG(wcagContext, true);
+    }
+
+    /**
+     * Validate WCAG properties on the current base page.
+     * @param wcagContext the current wcag context
+     * @param checkCache if set to true, the WCAGContext cache will be checked before starting validation and cancel it
+     *                   if any entry corresponds to the current BasePage.
+     */
+    protected void validateWCAG(WcagContext wcagContext, boolean checkCache)
+    {
+        if (!wcagContext.isWcagEnabled())
+        {
+            /*
+            Block wcag validation if it is not enabled, in all cases.
+             */
+            return;
+        }
```

Comment:
to be moved up there

Reply:
https://github.com/xwiki/xwiki-platform/pull/2092#discussion_r1115895145
I agreed with @michitux 's opinion, I think it's more important to include the check in the function for explicit validation calls in the UITests. 
However I did not give a wide enough scope (were set as `protected` instead of `public`).